src/MoleculeLookup.h

```diff
@@ -181,31 +176,53 @@ static uint GetConsensusMolBeta( const uint pStart,
 
   //array of indices for type Molecule, sorted by box and kind for
   //move selection
-  uint* molLookup;
-  uint molLookupCount;
+  std::vector<uint32_t> molLookup;
+  uint32_t molLookupCount;
   //index [BOX_TOTAL * kind + box] is the first element of that kind/box in
   //molLookup
   //index [BOX_TOTAL * kind + box + 1] is the element after the end
   //of that kind/box
-  uint* boxAndKindStart;
-  uint* boxAndKindSwappableCounts;
-  uint boxAndKindStartCount;
-  uint numKinds;
+  std::vector<uint32_t> boxAndKindStart;
+  std::vector<uint32_t> boxAndKindSwappableCounts;
+  uint32_t boxAndKindStartCount;
+  uint32_t numKinds;
   /* For consistent trajectory ordering across checkpoints */
-  bool restartFromCheckpoint;
-  uint32_t * originalMoleculeIndices, * permutedMoleculeIndices;
-
-  std::vector <uint> fixedMolecule;
-  std::vector <uint> canSwapKind; //Kinds that can move intra and inter box
-  std::vector <uint> canMoveKind; //Kinds that can move intra box only
-  int *molIndex; // stores the molecule index for global atom index
-  int *atomIndex; // stores the local atom index for global atom index
-  int *molKind; // stores the molecule kind for global atom index
-  int *atomKind; // stores the atom kind for global atom index
-  double *atomCharge; // stores the atom's charge for global atom index
+  std::vector<uint32_t> restartMoleculeIndices;
+  uint32_t restartedNumAtomsInBox[BOX_TOTAL];
+
+  std::vector <uint32_t> fixedMolecule;
+  std::vector <uint32_t> canSwapKind; //Kinds that can move intra and inter box
+  std::vector <uint32_t> canMoveKind; //Kinds that can move intra box only
+  std::vector<int32_t> molIndex; // stores the molecule index for global atom index
+  std::vector<int32_t> atomIndex; // stores the local atom index for global atom index
+  std::vector<int32_t> molKind; // stores the molecule kind for global atom index
+  std::vector<int32_t> atomKind; // stores the atom kind for global atom index
+  std::vector<double> atomCharge; // stores the atom's charge for global atom index
```

Comment:
I see here that you decided to use vector instead of array! Why?

Reply:
The Cereal library doesn't support serializing raw pointers.

Reply:
Restored these to dynamically allocated arrays. I used the binary_data() serialization function of Cereal, so I don't use vectors here at all.  They are allocated either in Init if Restart from Checkpoint is false, and in the CheckpointSetup if Restart from Checkpoint is true.